cdap-ui/app/cdap/components/PipelineNodeGraphs/PipelineNodeMetricsGraph.js

```diff
@@ -224,28 +299,41 @@ export default class PipelineNodeMetricsGraph extends Component {
       .subscribe(
         (res) => {
           let data = res.qid.series;
-          let recordsOut, recordsIn, recordsError;
+          let recordsOut, recordsIn, recordsError, recordsOutPorts;
+          recordsOut = recordsIn = recordsError = 0;
+          recordsOutPorts = {};
           let processTimeMetrics = {};
           data.forEach(d => {
             let metricName = d.metricName;
+            let dataValue = d.data[0].value;
```

Comment:
is `data[0]` guaranteed? if not, then i would suggest using objectQuery and default it to 0 as well, just in case

Reply:
I just added a commit to check where `d.data`'s length is greater than 0. I think if it does then the objects in the array are guaranteed to have `value` property.